product_variant_configurator/models/pricelist.py

```diff
@@ -1,214 +1,52 @@
 # -*- coding: utf-8 -*-
-# © 2015 Oihane Crucelaegui - AvanzOSC
-# © 2016 Pedro M. Baeza <pedro.baeza@tecnativa.com>
-# © 2016 ACSONE SA/NV
+# Copyright 2015 Oihane Crucelaegui - AvanzOSC
+# Copyright 2016 Pedro M. Baeza <pedro.baeza@tecnativa.com>
+# Copyright 2016 ACSONE SA/NV
+# Copyright 2017 David Vidal <david.vidal@tecnativa.com>
 # License AGPL-3 - See http://www.gnu.org/licenses/agpl-3
 
-from itertools import chain
-
-from openerp import models, fields, api, exceptions
+from odoo import api, models
 
 
 class ProductPricelist(models.Model):
     _inherit = 'product.pricelist'
 
     @api.multi
-    def _price_rule_get_multi(self, products_qty_partner):
-
-        self.ensure_one()
-        date = self._context.get('date', fields.Date.today())
-        uom_id = False
-        if self._context.get('uom'):
-            uom_id = self._context['uom']
+    def _compute_price_rule(self, products_qty_partner, date=False,
+                            uom_id=False):
+        """Overwrite for covering the case where templates are passed and a
+        different uom is used."""
+        if products_qty_partner[0][0]._name != "product.template":
+            # Standard use case - Nothing to do
+            return super(ProductPricelist, self)._compute_price_rule(
+                products_qty_partner, date=date, uom_id=uom_id,
+            )
+        if not uom_id and self.env.context.get('uom'):
+            ctx = dict(self.env.context)
+            # Remove uom context for avoiding the re-processing
+            uom_id = ctx.pop('uom')
+            self = self.with_context(ctx)
```

Comment:
No, David, he is saying that use another variable name, like `obj`: `obj = self.with_context(ctx)`